network.go

```diff
@@ -50,137 +40,181 @@ func (nw *netWatcher) Results() (MeasurementsMap, error) {
 				log.Errorf("[NET] net_interface_exclude_regex regexp '%s' compile error: %s", reString, err.Error())
 				continue
 			}
-			netInterfaceExcludeRegexCompiled = append(netInterfaceExcludeRegexCompiled, re)
+			nw.netInterfaceExcludeRegexCompiled = append(nw.netInterfaceExcludeRegexCompiled, re)
 		}
 	}
 
-	for _, netIf := range interfaces {
-		if isExcluded, cacheExists := nw.ExcludedInterfaceCache[netIf.Name]; cacheExists {
-			if isExcluded {
-				log.Debugf("[NET] interface excluded: %s", netIf.Name)
-				continue
-			}
-		} else {
-			isExcluded := false
-
-			if nw.cagent.NetInterfaceExcludeLoopback {
-				loopback := false
-				for _, flag := range netIf.Flags {
-					if flag == "loopback" {
-						loopback = true
-						break
-					}
-				}
-
-				if loopback {
-					isExcluded = true
-				}
-			}
+	return nw.netInterfaceExcludeRegexCompiled
+}
 
-			if !isExcluded && nw.cagent.NetInterfaceExcludeDisconnected {
-				up := false
-				for _, flag := range netIf.Flags {
-					if flag == "up" {
-						up = true
-						break
-					}
-				}
-				if !up {
-					isExcluded = true
-				}
-			}
+func isInterfaceLoobpack(netIf *utilnet.InterfaceStat) bool {
+	for _, flag := range netIf.Flags {
+		if flag == "loopback" {
+			return true
+		}
+	}
 
-			if !isExcluded {
-				for _, excludedIf := range nw.cagent.NetInterfaceExclude {
-					if strings.EqualFold(netIf.Name, excludedIf) {
-						isExcluded = true
-						break
-					}
-				}
-			}
+	return false
+}
 
-			if !isExcluded {
-				for _, re := range netInterfaceExcludeRegexCompiled {
-					if re.MatchString(netIf.Name) {
-						isExcluded = true
-						break
-					}
-				}
-			}
+func isInterfaceDown(netIf *utilnet.InterfaceStat) bool {
+	for _, flag := range netIf.Flags {
+		if flag == "up" {
+			return false
+		}
+	}
+
+	return true
+}
 
-			nw.ExcludedInterfaceCache[netIf.Name] = isExcluded
+func (nw *netWatcher) isInterfaceExcludedByName(netIf *utilnet.InterfaceStat) bool {
+	for _, excludedIf := range nw.cagent.NetInterfaceExclude {
+		if strings.EqualFold(netIf.Name, excludedIf) {
+			return true
+		}
+	}
+	return false
+}
 
-			if isExcluded {
+func (nw *netWatcher) isInterfaceExcludedByRegexp(netIf *utilnet.InterfaceStat) bool {
+	for _, re := range nw.InterfaceExcludeRegexCompiled() {
+		if re.MatchString(netIf.Name) {
+			return true
+		}
+	}
+	return false
+}
+
+func (nw *netWatcher) ExcludedInterfacesByNameMap(allInterfaces []utilnet.InterfaceStat) map[string]struct{} {
```

Comment:
Maybe remove the Map from the function name?
There is no other `ExcludedInterfacesByName...` I think and we don't have to repeat the return type

Reply:
The main reason for this function is to return the map(to be able to check interface by name). Because the list(slice) itself was already available.

Reply:
I see, so it converts the slice into a map. Then maybe rename to `ConvertExcludedInterfacesfromSliceToMap` Also quite verbose but more clear what's happening.

Reply:
if a function called Convert it should accept the slice to convert. But in this case it is filter interfaces and provide a map of filtered out interfaces... BTW `ExcludedInterfacesByName` is ok in fact, because it contains `ByName` suffix and the argument `allInterfaces` is clear enough

Reply:
> But in this case it is filter interfaces and provide a map of filtered out interfaces

I see. These are the details that are perfectly placed in a short comment in front of the function :)